src/main/java/org/symphonyoss/symphony/messageml/elements/TextField.java

```diff
@@ -199,18 +201,53 @@ private void buildAttributeFromPresentationML(org.w3c.dom.Node item) throws Inva
 
   private void validateMinAndMaxLengths() throws InvalidInputException {
     Integer maxLength = getAttributeAsInteger(MAXLENGTH_ATTR);
-    if (isLengthIsOutOfRange(maxLength)) {
+    if (isLengthOutOfRange(maxLength)) {
       throw new InvalidInputException(getLengthErrorMessage(MAXLENGTH_ATTR));
     }
 
     Integer minLength = getAttributeAsInteger(MINLENGTH_ATTR);
-    if (isLengthIsOutOfRange(minLength)) {
+    if (isLengthOutOfRange(minLength)) {
       throw new InvalidInputException(getLengthErrorMessage(MINLENGTH_ATTR));
     }
 
-    if (minLength != null && maxLength != null && minLength > maxLength) {
+    minLength = getDefaultValueIfCurrentIsNull(minLength, MIN_ALLOWED_LENGTH);
+    maxLength = getDefaultValueIfCurrentIsNull(maxLength, MAX_ALLOWED_LENGTH);
+
+    if (isMinAndMaxLengthCombinationValid(maxLength, minLength)) {
       throw new InvalidInputException("The attribute \"minlength\" must be lower than the \"maxlength\" attribute");
     }
+
+    if (textFieldHasInitialValue()) {
+      String initialValue = getTextFieldInitialValue();
+      if (isTextSmallerThanMinLength(minLength, initialValue) || isTextBiggerThanMaxLength(maxLength, initialValue)) {
+        throw new InvalidInputException(
+            format("The length of a text-field's initial value must be between %s and %s", minLength, maxLength));
```

Comment:
"The length of a text-field's initial value" this is weird. I had to re-read it to understand and almost commented thinking you were not mentioning a single instance of the text field. I would use "this" to mention a particular text-field and not the " 's ".